src/main/java/org/osc/manager/rest/server/api/DomainApis.java

```diff
@@ -40,76 +49,333 @@
             PolicyEntity entity) {
 
-        logger.info("Updating Policy Entity ...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Updating for Policy Entity Id...:" + policyId);
+
+        return this.txControl.required(new Callable<PolicyEntity>() {
+
+            @Override
+            public PolicyEntity call() throws Exception {
+
+                CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+
+                CriteriaQuery<PolicyEntity> query = criteriaBuilder.createQuery(PolicyEntity.class);
+                Root<PolicyEntity> r = query.from(PolicyEntity.class);
+                query.select(r).where(criteriaBuilder.and(criteriaBuilder.equal(r.get("domain").get("id"), domainId)),
+                        criteriaBuilder.equal(r.get("id"), policyId));
+
+                List<PolicyEntity> result = em.createQuery(query).getResultList();
+                if (result.isEmpty()) {
+                    throw new Exception("domain or the policy Entity does not exists...");
+                    //TODO - to add RETURN 404 error:Sudhir
+                }
+                result.get(0).setName(entity.getName());
+                em.persist(result.get(0));
+                return result.get(0);
+            }
+        });
     }
 
     /**
      * Deletes the Policy for a given domain and the policy
      *
      * @return - deleted policy
      */
-    @Path("/{domainid}/policies/{policyid}")
+    @Path("/{domainId}/policies/{policyId}")
     @DELETE
-    public PolicyEntity deletePolicy(@PathParam("domainid") Long domainid, @PathParam("policyid") Long policyid,
+    public void deletePolicy(@PathParam("domainId") Long domainId, @PathParam("policyId") Long policyId,
             PolicyEntity entity) {
 
-        logger.info("Deleting Policies Entity ...");
-        //TODO Sudhir: Add db calls here
-        return null;
+        logger.info("Deleting for Policies Entity Id...:" + policyId);
+
+        this.txControl.required(new Callable<Void>() {
+
+            @Override
+            public Void call() throws Exception {
+
+                DomainEntity result = em.find(DomainEntity.class, domainId);
+                if (result == null) {
+                    return null;
+                }
+                PolicyEntity policyresult = em.find(PolicyEntity.class, policyId);
+                if (policyresult == null) {
+                    return null;
+                }
+                em.remove(policyresult);
+                return null;
+            }
+        });
     }
 
     /**
      * Lists the Policy Id's for a given domain
      *
      * @return - Policy Id's
      */
-    @Path("/{domainid}")
+    @Path("/{domainId}/policies")
     @GET
-    public List<String> getPolicyIds(@PathParam("domainid") Long domainid) {
+    public List<String> getPolicyIds(@PathParam("domainId") Long domainId) {
```

Comment:
@sudhirappaji isnt majority of the code in this method and other methods the same as [IsmPolicyApi.java](https://github.com/opensecuritycontroller/security-mgr-sample-plugin/blob/master/src/main/java/org/osc/manager/ism/api/IsmPolicyApi.java#L79)

Why are we duplicating the code? @emanoelxavier thoughts?

Reply:
during early part of the plugin discussion, it was set as a requirement that REST implementation should not use the plugin API. The reason mentioned was that REST and plugin should be independent and any change in plugin code should not have impact on REST implementation. This was the reason why plugin api's are not used in REST implementation.

Reply:
I had discussion with Emanoel and agreed that this shall be implemented for upcoming user stories and will handle the comment as part of other user-stories. will use the public methods for example in ismpolicyapi in the rest implementation. This way it won't break when the SDK API changes.